### PR TITLE
Implement IETF Referenced Token Status claims

### PR DESCRIPTION
**Pull request by eelanagaraj**
### Description
- Implements Status claim included in the referenced token (`ReferencedTokenStatus`), as described in the IETF Token Status spec
  - Adds all nested claims
  - Adds convenience trait implementations so that this can be used in a CWT claims set

### Tested
- Added a unit test for encoding/decoding
- Added a unit test case for the nested type

### CODE REVIEW

File: src/cwt/claim/ietf_token_status.rs
Comment (timothee-haudebourg) [inline src/cwt/claim/ietf_token_status.rs:61]: Why do you need to implement `Claim` for `StatusListReference`? I know they use the term "claim" in the spec to talk about `idx` and `uri`, but I think the only real CWT claim is `status_list` (65535), is it not?

File: src/cwt/claim/ietf_token_status.rs
Comment (timothee-haudebourg) [inline src/cwt/claim/ietf_token_status.rs:92]: Same comment here, I don't think that's necessary.

File: src/cwt/claim/ietf_token_status.rs
Comment (timothee-haudebourg) [inline src/cwt/claim/ietf_token_status.rs:66]: If my comment above is right, I think you can get rid of the conversion impls and just rely on `serde` to serialize/deserialize the `ReferenceTokenValue`.

File: src/cwt/claim/ietf_token_status.rs
Comment (timothee-haudebourg) [inline src/cwt/claim/ietf_token_status.rs:115]: Same comment here, I don't think that's necessary.

File: src/cwt/claim/ietf_token_status.rs
Comment (timothee-haudebourg) [inline src/cwt/claim/ietf_token_status.rs:28]: I know a great URI library called [`iref`](https://github.com/timothee-haudebourg/iref) that provides a nice [`UriBuf`](https://docs.rs/iref/latest/iref/uri/struct.UriBuf.html) type you can use instead of `StatusListUri` (or inside it if you want to keep a newtype). It has `serde` support with the `serde` feature. It's already used everywhere in ssi. I may be biased here :smile:

File: src/cwt/claim/ietf_token_status.rs
Comment (timothee-haudebourg) [inline src/cwt/claim/ietf_token_status.rs:23]: It depends or whether or not I'm right about my `Claim` implementation comments below, but if so you don't really need a newtype here.

**Review by timothee-haudebourg — COMMENTED** [review]
You could remove a lot of stuff by not considering `StatusListReference`, `StatusListIndex` and `StatusListUri` as CWT claims. My interpretation of the spec is that only `ReferencedTokenStatus` is a CWT claims (you will never see a CWT out in the wild with `idx` as a top-level claim). I don't have strong opinions on the `StatusListIndex` and `StatusListUri` newtypes, but you could also get rid of them. Overall it's a matter of interpretation of the spec whether or not you want to make changes.

File: src/cwt/claim/ietf_token_status.rs
Comment (eelanagaraj) [reply src/cwt/claim/ietf_token_status.rs:61]: Oh maybe I misinterpreted that 🤔 I read it thinking they are all supposed to be separate claims but maybe that's not the case. @cobward do you know/have thoughts?

File: src/cwt/claim/ietf_token_status.rs
Comment (eelanagaraj) [reply src/cwt/claim/ietf_token_status.rs:28]: yeah wanted to get your thoughts on this -- for now I left it as open as possible to not introduce an additional dependency & allow consumers of this library to choose whatever underlying url/uri representation they want, but maybe going with iref makes sense here 🤔

Comment (cobward) [reply src/cwt/claim/ietf_token_status.rs:28]: I would hold off on this until we decide to merge this into `ssi`.

File: src/cwt/claim/ietf_token_status.rs
Comment (cobward) [reply src/cwt/claim/ietf_token_status.rs:61]: +1 to what @timothee-haudebourg said (except there's a slight error in the name). `status` (65535) is the only claim. Extracting a status should yield the following structure (CBOR-diagnostic notation):
```
      {
          "status_list": {
            "idx": "0",
            "uri": "https://example.com/statuslists/1"
          }
      }
 ```
 
 So I would suggest having a structs like this:
 ```rust
 pub struct Status {
      status_list: StatusList,
 }
 
 pub struct StatusList {
      idx: u32,
      uri: String,
 }
 
 impl Claim for Status { ... }
 ```

Comment (cobward) [reply src/cwt/claim/ietf_token_status.rs:61]: Possible even something like this to handle spec extensions gracefully:

```rust
pub struct Status {
     pub status_list: StatusList,
     pub extra: BTreeMap<String, serde_cbor::Value>,
}
```

**Review by timothee-haudebourg — APPROVED** [review]
Looks good to me!

File: src/cwt/claim/ietf_token_status.rs
Comment (eelanagaraj) [reply src/cwt/claim/ietf_token_status.rs]: ahh I missed the spec extension comment earlier -- left it out for now but can add it in if needed!

File: src/cwt/claim/ietf_token_status.rs
Comment (cobward) [inline src/cwt/claim/ietf_token_status.rs:15]: Rename to `StatusListInfo` to distinguish from the `StatusList` structure defined in section 4.2.
```suggestion
pub struct StatusListInfo {
```

**Review by cobward — APPROVED** [review]

**Pull request merged**
